script.js

```diff
@@ -189,17 +216,78 @@ function handleData([sheetsText, statesGeoJson]) {
     }
   });
 
+  countiesGeoJson = {
+    type: "FeatureCollection",
+    features: countiesCartoData.rows.map(
+      ({ the_geom, state, county }, index) => {
+        const properties = countiesData.filter(
+          row =>
+            row.municipality
+              .replace(" County", "")
+              .replace("Miami-Dade", "Dade")
+              .replace("County of ", "")
+              .replace(" (unincorporated)", "")
+              .replace(" County", "") === county &&
+            row.state.replace("Illinoise", "Illinois") === state
+        );
+
+        return {
+          type: "Feature",
+          id: index,
+          properties,
+          geometry: JSON.parse(the_geom)
+        };
+      }
+    )
+  };
+
+  console.log(countiesGeoJson);
+
   // add both the states layer and localities layer to the map
   // and save the layer output
   const states = handleStatesLayer(statesGeoJson);
+  const counties = handleCountiesCartoLayer(countiesGeoJson);
   const localities = handleLocalitiesLayer(localitiesGeoJson);
-
+  
   // add layers to layers control
   layersControl
-    .addOverlay(localities, "Cities/Counties")
+    .addOverlay(localities, "Cities")
+    .addOverlay(counties, "Counties")
     .addOverlay(states, "States");
 }
 
+function createCartoUri(rows) {
+  const countiesData = rows.filter(row => row.admin_scale === "County");
+
+  const predicateString = countiesData
+    .map(({ state, municipality }) => {
+      return {
+        state,
+        municipality
+      };
+    })
+    .reduce(
+      (acc, row) =>
+        `${acc} county LIKE '${row.municipality
+          .replace(" County", "")
+          .replace("Miami-Dade", "Dade")
+          .replace("County of ", "")
+          .replace(" (unincorporated)", "")
+          .replace(" County", "")}' AND state LIKE '${row.state.replace(
+          "Illinoise",
+          "Illinois"
+        )}' OR`,
+      "WHERE"
+    )
+    .slice(0, -3);
+
+  cartoQuery = `SELECT county, state, ST_AsGeoJSON(the_geom) as the_geom from public.us_county_boundaries ${predicateString}`;
+
+  console.log(cartoQuery);
+
+  return `https://ampitup.carto.com/api/v2/sql?q=${cartoQuery}`;
```

Comment:
@timhitchins note that you can pass an additional query param to the SQL API to get data back as GeoJSON:

```js
  return `https://ampitup.carto.com/api/v2/sql?q=${cartoQuery}&format=GeoJSON`;
```
[link to SQL API docs](https://carto.com/developers/sql-api/reference/#tag/Single-SQL-Statement)

if combined with the spatial join this would replace the need for manually creating the GeoJSON client side.

Reply:
Haha.  This is waaaay better.